skbio/stats/composition.py

```diff
@@ -489,13 +527,36 @@ def clr(mat):
     array([-0.79451346,  0.30409883,  0.5917809 , -0.10136628])
 
     """
-    mat = closure(mat)
-    lmat = np.log(mat)
-    gm = lmat.mean(axis=-1, keepdims=True)
-    return (lmat - gm).squeeze()
+    axis = -1
+    # xp is the namespace wrapper for different array libraries
+    # NOTE: the following (try:) may reduce the efficiency but to
+    # keep backward compatibility when the input is a list
+    try:
+        xp = aac.array_namespace(mat)
+    except Exception as e:
+        mat = np.asarray(mat)
+        xp = aac.array_namespace(mat)
 
 
-def clr_inv(mat):
+    if validate:
+        # assert from closure() while it is removed in latest version
+        if xp.any(mat < 0):
+            raise ValueError("Cannot have negative proportions")
+        if mat.ndim > 2:
+            raise ValueError("Input matrix can only have two dimensions or less")
+
+        # check if the input is a composition
+        _composition_check(mat)
+
+    original_shape = mat.shape
+    # squeeze the singleton dimensions
```

Comment:
@songyesog2000 I think that `squeeze` is part of the Python API standard ([source](https://data-apis.org/array-api/latest/API_specification/generated/array_api.squeeze.html)) and it automatically removes all singleton dimensions. So the following line can just be `mat = xp.squeeze(mat)`?